src/views/Alteration.vue

```diff
@@ -182,16 +182,27 @@ export default class Alteration extends Mixins(CommonMixin, LegalApiMixin, Filin
     Vue.nextTick(() => this.setHaveChanges(false))
   }
 
-  /** Fetch Business Snapshot */
-  private async fetchBusinessSnapshot (): Promise<BusinessSnapshotIF[]> {
-    return Promise.all([
-      this.getBusinessData(),
-      this.getBusinessData(BusinessDataTypes.TRANSLATIONS),
-      this.getBusinessData(BusinessDataTypes.ADDRESSES),
-      this.getBusinessData(BusinessDataTypes.DIRECTORS),
-      this.getBusinessData(BusinessDataTypes.SHARE_CLASSSES),
-      this.getContactInfo()
+  /** Fetches the business snapshot. */
+  private async fetchBusinessSnapshot (): Promise<BusinessSnapshotIF> {
+    const items = await Promise.all([
+      this.fetchBusinessInfo(),
+      this.fetchContactPoint(),
+      this.fetchIncorporationAddress(),
+      this.fetchNameTranslations(),
+      this.fetchOrgPersons(),
+      this.fetchShareStructure()
     ])
+
+    if (items.length !== 6) throw new Error('Failed to fetch business snapshot')
+
+    return {
+      businessInfo: items[0],
+      contactPoint: items[1],
+      incorporationAddress: items[2],
+      nameTranslations: items[3],
+      orgPersons: items[4],
+      shareStructure: items[5]
+    }
```

Comment:
Business snapshot is no longer an array with each element containing one of these sub-objects :)